src/garage/_dtypes.py

```diff
@@ -167,3 +167,199 @@ def __new__(cls, env_spec, observations, actions, rewards, terminals,
         return super().__new__(TrajectoryBatch, env_spec, observations,
                                actions, rewards, terminals, env_infos,
                                agent_infos, lengths)
+
+class SamplesBatch(
+        collections.namedtuple('SamplesBatch', [
+            'env_spec',
+            'observations',
+            'actions',
+            'rewards',
+            'next_observations',
+            'terminals',
+            'env_infos',
+            'agent_infos',
+        ])):
+    # pylint: disable=missing-return-doc, missing-return-type-doc, missing-param-doc, missing-type-doc  # noqa: E501
+    r"""A tuple representing a batch of whole trajectories.
+
+    A :class:`SamplesBatch` represents a batch of samples
+    produced when one or more agents interacts with one or more environments.
+
+    Attributes:
+        env_spec (garage.envs.EnvSpec): Specification for the environment from
+            which this data was sampled.
+        observations (numpy.ndarray): A numpy array of shape
+            :math:`(N \bullet [T], O^*)` containing the (possibly
+            multi-dimensional) observations for all time steps in this batch.
+            These must conform to :obj:`env_spec.observation_space`.
+        actions (numpy.ndarray): A  numpy array of shape
+            :math:`(N \bullet [T], A^*)` containing the (possibly
+            multi-dimensional) actions for all time steps in this batch. These
+            must conform to :obj:`env_spec.action_space`.
+        rewards (numpy.ndarray): A numpy array of shape
+            :math:`(N \bullet [T])` containing the rewards for all time steps
+            in this batch.
+        terminals (numpy.ndarray): A boolean numpy array of shape
+            :math:`(N \bullet [T])` containing the termination signals for all
+            time steps in this batch.
+        env_infos (dict): A dict of numpy arrays arbitrary environment state
+            information. Each value of this dict should be a numpy array of
+            shape :math:`(N \bullet [T])` or :math:`(N \bullet [T], S^*)`.
+        agent_infos (numpy.ndarray): A dict of numpy arrays arbitrary agent
+            state information. Each value of this dict should be a numpy array
+            of shape :math:`(N \bullet [T])` or :math:`(N \bullet [T], S^*)`.
+            For example, this may contain the hidden states from an RNN policy.
+
+    Raises:
+        ValueError: If any of the above attributes do not conform to their
+            prescribed types and shapes.
+
+    """
+    __slots__ = ()
+
+    def __new__(cls, env_spec, observations, actions, rewards, next_observations,
+                 terminals, env_infos, agent_infos):  # noqa: D102
```

Comment:
Is it that the only differences between `SamplesBatch` and `TrajectoryBatch` are 
- `SamplesBatch` incorporates `next_observations`
- But does not incorporate `lengths`?

Reply:
the samples in a sample batch don't necessarily have any meaningful time relationship.